src/main/java/com/luismateoh/gymcrm/dto/TrainerDTO.java

```diff
@@ -1,4 +1,22 @@
 package com.luismateoh.gymcrm.dto;
 
-public record TrainerDTO(String firstName, String lastName, String specialization) {
+import jakarta.validation.constraints.NotNull;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class TrainerDTO {
+    private Long id;
+
+    @NotNull(message = "User cannot be null")
+    private UserDTO user;
+
+    @NotNull(message = "Specialization cannot be null")
+    private TrainingTypeDTO specialization;
+
+    @Override
+    public String toString() {
+        return "Trainer -> " + "ID: " + id + ", Specialization: " + specialization.getTrainingTypeName() + "\n" + user + "\n";
```

Comment:
String interpolation is  a better approach. Sample String.format("String %s in %s with some %s examples.", first, second, second)